navjson/navjson.go

```diff
@@ -0,0 +1,58 @@
+// Package navjson provides a `nav.json` parser and validator.
+package navjson
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// NavFile represends a file/path association in a `nav.json` document.
+type NavFile struct {
+	Path  string    `json:"path"`
+	Files []NavFile `json:"files,omitempty"`
+}
+
+// NavPackage represents a list of packages in a `nav.json` document. This is only used for monorepos.
+type NavPackage map[string]string
+
+// NavJSON represets a `nav.json` document.
+type NavJSON struct {
+	Root             string     `json:"root"`
+	AssetRoot        string     `json:"assetRoot"`
+	SkipMenuOrdering bool       `json:"skipMenuOrdering,omitempty"`
+	Packages         NavPackage `json:"packages,omitempty"`
+	Files            []NavFile  `json:"files"`
+}
+
+// Parse attempts to parse the given data into a NavJSON object.
+func Parse(data string) (NavJSON, error) {
+	j := NavJSON{}
+	b := []byte(data)
+	err := json.Unmarshal(b, &j)
+	if err != nil {
+		return j, err
+	}
+
+	if j.Root == "" {
```

Comment:
instead of checking each `prop`, can we do equivalent of `Object.keys(my_struct).map(k => ...validate...)`

Reply:
Nope. Golang doesn't have generics, so this isn't possible.